pinot-common/src/main/java/org/apache/pinot/common/utils/FileUtils.java

```diff
@@ -64,4 +70,63 @@ public static void transferBytes(FileChannel src, long position, long count, Fil
       count -= numBytesTransferred;
     }
   }
+
+  /**
+   * Close a collection of {@link Closeable} resources
+   * This is a utility method to help release multiple {@link Closeable}
+   * resources in a safe manner without leaking.
+   * As an example if we have a list of Closeable resources,
+   * then the following code is prone to leaking 1 or more
+   * subsequent resources if an exception is thrown while
+   * closing one of them.
+   *
+   * for (closeable_resource : resources) {
+   *   closeable_resource.close()
+   * }
+   *
+   * The helper methods provided here do this safely
+   * while keeping track of exception(s) raised during
+   * close() of each resource and still continuing to close
+   * subsequent resources.
+   * @param closeables collection of resources to close
+   * @throws IOException
+   */
+  public static void close(@Nonnull Iterable<? extends Closeable> closeables)
```

Comment:
We follow the convention of only annotating nullable argument and tread non-annotated argument as non-null. (We did annotate non-null before and found there are too many annotations and Nullable and Nonnull look very similar :-P)

Reply:
done